clients/gtest/rocblas_test.cpp

```diff
@@ -21,19 +21,24 @@
  * ************************************************************************ */
 
 #include "rocblas_test.hpp"
-#include "client_utility.hpp"
 
-#include <cstdlib>
-#include <exception>
-#include <regex>
 #ifdef WIN32
+// Must include windows.h before dependent headers.
+// Specifically this must be before client_utility.hpp uses `#pragma GCC poison`
+// to poison 'ctime' and 'abort', as those are used in standard library headers.
 #include <windows.h>
```

Comment:
If no Cryptography, DDE, RPC, Shell, and Windows sockets are required, you should prepend `#include <Windows.h>` with `#define WIN32_LEAN_AND_MEAN` as such:
```C++
#idefine WIN32_LEAN_AND_MEAN
#include <Windows.h>
```
The correct case of the `Windows.h` is with the capital letter (for MSFT)
There are more flags to minimize the footprint. Read it more here: https://learn.microsoft.com/en-us/windows/win32/winprog/using-the-windows-headers
But usually `WIN32_LEAN_AND_MEAN` is enough.

Reply:
The LEAN define is added in our toolchain-windows.cmake where used with the original fail fast design pattern.  This is a new windows build pipeline which may not use that, I will need to confirm that sometime, but for now if both builds succeed this is fine to merge.

Reply:
TheRock currently sets the same defines for each ROCm sub-project with a common toolchain file constructed here: https://github.com/ROCm/TheRock/blob/b5412338880fed74976b2f4408e0e06cbccc683f/cmake/therock_subproject.cmake#L1111-L1119 . I did not need to add the `WIN32_LEAN_AND_MEAN` define for build correctness, but agree that it could be beneficial to add. Definitely worth doing a scrub through and setting more build size and performance flags, but might need to customize defines/flags per sub-project as in the current project-specific toolchain files.

Reply:
Well that explains the error you see vs. what we get in our windows CI.   It is not for correctness but for fewer dependencies being included.  Some component other than rocBLAS may use the dependencies @apwojcik listed but probably not if they copied the rocBLAS toolchain (which was done first).